src/aspire/source/image.py

```diff
@@ -339,36 +474,69 @@ def set_metadata(self, metadata_fields, values, indices=None):
             values should either be a scalar or a vector of length equal to the total number of images, |self.n|.
         :return: On return, the metadata associated with the specified indices has been modified.
         """
-        # Convert a single metadata field into a list of single metadata field, since that's what the 'columns'
-        # argument of a DataFrame constructor expects.
+
```

Comment:
Is `values` supposed to be one- or two-dimensional (the fact that `metadata_fields` can be a vector suggests both, but docstring is not clear here).

Reply:
I think values can be 0, 1, or 2d.

Looking closer now, Im not sure the behavior when using multiple fields in conjunction with values would work for the dense case (where we dont want to assign the same values to each field.).  Will look a little more.

Reply:
Ok. Maybe just make an issue and leave this for now (from what I understand, the behavior is unchanged from before).

Reply:
Sure, https://github.com/ComputationalCryoEM/ASPIRE-Python/issues/924, thanks.